pkg/services/rpcsrv/util.go

```diff
@@ -18,3 +24,46 @@ func checkInt32(i int) error {
 	}
 	return nil
 }
+
+func (c notificationComparatorFilter) EventID() neorpc.EventID {
+	return c.id
```

Comment:
`c.id` field is not needed, remove it. `notificatoinEventComparator` works only with `ContainedNotificatoinEvent`, hence you may return `neorpc.NotificationEventID` here.